timemachine/fe/mcgregor.py

```diff
@@ -246,7 +254,7 @@ def mcs(
         core_array = np.array(sorted(core))
         all_cores.append(core_array)
 
-    return all_cores, mcs_result.all_marcs
+    return all_cores, mcs_result.all_marcs, total_nodes_visited
```

Comment:
Should we also return the total number of cores created? Not sure if that would also be a useful stat

Reply:
Not sure; I guess I'd be inclined to add it as we need it, especially since we'd probably want to introduce a dataclass or something to hold the diagnostics if we return more than one.

Reply:
Any reason not to add the dataclass now? Seems like having these diagnostics sooner rather than later would be handy going forward.

Reply:
Added an `MCSDiagnostics` dataclass in https://github.com/proteneer/timemachine/pull/1231/commits/d576f2f2477a514d9c90202f8283baf22a51eeb5. This has the additional benefit that new fields can be added without breaking unpickling of older results (assuming new fields default to `None`).

Reply:
Sweet, that will be nice. Should we add the number of visited cores now that we have the dataclass?

Reply:
Added additional fields in https://github.com/proteneer/timemachine/pull/1231/commits/4d6325a198af06c0c04b816b88fb67938b492c95